app/src/main/java/org/wikipedia/readinglist/database/ReadingListDbHelper.java

```diff
@@ -240,12 +266,52 @@ void addPagesToList(@NonNull SQLiteDatabase db, @NonNull ReadingList list, @NonN
         return addedTitles;
     }
 
+    public List<String> movePagesToListIfNotExist(long sourceReadingListId, @NonNull ReadingList list, @NonNull List<PageTitle> titles) {
+        SQLiteDatabase db = getWritableDatabase();
+        db.beginTransaction();
+        List<String> movedTitles = new ArrayList<>();
+        try {
+            for (PageTitle title : titles) {
+                if (getPageByTitle(db, list.id(), title) != null) {
+                    continue;
+                }
```

Comment:
This will behave incorrectly.  If you move a title from a reading list to another reading list that already has that title, the title won't be removed from the first reading list.

Reply:
Good catch! I noticed that the message `All good! %1$s already contains %2$s.` will no longer be shown with the new logic because the `movedTitles` will not be empty. 

If we still want to show the message above, we'll need to call `getPageByTitle()` in the `MoveToReadingListDialog` for the message. Should we do that?

Reply:
I think this is fine as is.